### ISSUE TITLE
;gitlog fails if a problem gets renamed after being added to challenge table

### ISSUE THREAD

**Issue by Denjell** [cheran-senthil/TLE#460]
;gitgud gave me the following problem: https://codeforces.com/contest/675/problem/D
At that time it was only available in russian language and named Дерево.
After i solved it a friend got the problem translated and now its known under a different name. 
;gitlog now fails with an exception since it looks into problem cache using the old saved name.

The whole exception: 
06-06-2021 23:35:33:ERROR:tle.util.discord_common:Ignoring exception in command gitlog:
Traceback (most recent call last):
  File "/home/denjell/.cache/pypoetry/virtualenvs/tle-uDqy6Ktp-py3.7/lib/python3.7/site-packages/discord/ext/commands/core.py", line 85, in wrapped
    ret = await coro(*args, **kwargs)
  File "/home/denjell/TLE/tle/cogs/codeforces.py", line 326, in gitlog
    pages = [make_page(chunk, score) for chunk in paginator.chunkify(data, 10)]
  File "/home/denjell/TLE/tle/cogs/codeforces.py", line 326, in <listcomp>
    pages = [make_page(chunk, score) for chunk in paginator.chunkify(data, 10)]
  File "/home/denjell/TLE/tle/cogs/codeforces.py", line 312, in make_page
    log_str = '\n'.join(make_line(entry) for entry in chunk)
  File "/home/denjell/TLE/tle/cogs/codeforces.py", line 312, in <genexpr>
    log_str = '\n'.join(make_line(entry) for entry in chunk)
  File "/home/denjell/TLE/tle/cogs/codeforces.py", line 302, in make_line
    problem = cf_common.cache2.problem_cache.problem_by_name[name]
KeyError: 'Дерево'

**Comment by Denjell** [comment]
Fixed it manually in DB.

**Comment by algmyr** [comment]
The issue from what I remember is that cf doesn't give us any sensible id for problems, so we have no choice but using something like name. Unless we create our own indexing system in the db, which must be resilient to recreating the db (the same every time), which seems like pain

**Comment by bjorn-martinsson** [comment]
Would it be possiblee to make something like unmagic but for problems?
```
;handle unmagic 

Updates handle of the calling user if they have changed handles
(typically new year's magic)
```

**Comment by Denjell** [comment]
I think its fine to leave the issue unadressed. I only saw one case and it was somehow triggered to my actions. I got a problem that had no english translation as gitgud (with russian title) and asked Maksim1744 for a translation so that i can solve it. He also got the translation added to CF and this way the title of my gitgud and the title in the problem cache diverged. 
I think the problem occurs very seldom so we can live with it.
And yeah its a shame that CF still has no problem and user ids in the API. Would make shit so much easier.
